fe/fe-core/src/main/java/com/starrocks/planner/NestLoopJoinNode.java

```diff
@@ -59,53 +58,54 @@ public void buildRuntimeFilters(IdGenerator<RuntimeFilterId> generator, Descript
         if (!joinOp.isInnerJoin() && !joinOp.isLeftSemiJoin() && !joinOp.isRightJoin() && !joinOp.isCrossJoin()) {
             return;
         }
-        SessionVariable sessionVariable = ConnectContext.get().getSessionVariable();
-        PlanNode buildStageNode = this.getChild(1);
+        if (!ConnectContext.get().getSessionVariable().isEnableCrossJoinRuntimeFilter()) {
+            return;
+        }
+
         List<Expr> conjuncts = new ArrayList<>(otherJoinConjuncts);
         conjuncts.addAll(getConjuncts());
         for (int i = 0; i < conjuncts.size(); i++) {
             Expr expr = conjuncts.get(i);
-            if (canBuildFilter(expr)) {
+            if (expr.getChildren().size() == 2) {
```

Comment:
expr that has more than 2 children should also can support this feature,  so the condition should be
expr's children can be split into two group,  one group only references to SlotRefs of build side, the other references to SlotRefsof probe side.